lib/ios/RNNTopBarOptions.m

```diff
@@ -167,7 +167,18 @@ - (void)applyOn:(UIViewController*)viewController {
 	}
 	
 	viewController.navigationItem.hidesBackButton = [self.backButtonHidden boolValue];
-	
+    
+    if (self.barStyle) {
```

Comment:
Would it make more sense to pass in `UIBarStyleBlack` or `UIBarStyleBlackOpaque` instead of doing this loose string comparison? Seems more scalable that way.

Reply:
```
viewController.navigationController.navigationBar.barStyle = self.barStyle || UIBarMetricsDefault;

Reply:
you can't pass enums across the bridge.. you can fake it by passing in numbers.. but theres no such thing in actuality. check out how they do this guy. https://github.com/nfl/react-native-navigation/blob/v2/lib/ios/RNNRootViewController.m#L138-L144 that's what i mimicked this after... also. that `||` null coercing operator doesn't work on strictly typed languages bro.. lol

Reply:
LOL .. Swift <> JS
I was going to say that maybe we should stick to the integer number so we use the same underlying value as Apple does. So when they deprecate something we should still be good to go, like here:
```objective-c
UIBarStyleBlack            = 1,
UIBarStyleBlackOpaque      = 1, // Deprecated. Use UIBarStyleBlack
```
BUT, string is more legible .. so I don't really have a preference. If that guy is using string, we can keep the pattern.

Reply:
yeah.. you can create an enum that matches it on es6 but it makes the usage a lot more opaque..
like
```
import {someEnum} from "react-native-navigation";
...
static get options(): * {
    topBar: {
        statusBar: someEnum.default
    }
}
```
seems overkill.. especially since the way they define their API is via string values.

Reply:
yeah, I see why. Okay, LGTM